recipe-client-addon/lib/RecipeRunner.jsm

```diff
@@ -81,6 +81,20 @@ this.RecipeRunner = {
       await ClientEnvironment.getClientClassification();
     }
 
+    // Run pre-execution hooks. If a hook fails, we don't run recipes with that
+    // action to avoid inconsistencies.
+    const actions = await NormandyApi.fetchActions();
+    for (const action of Object.values(actions)) {
+      try {
+        await this.runActionCallback(action, "preExecution");
```

Comment:
Yes. I intend to do a documentation pass once we get the MVP for experiments ready for mozilla-central.